src/pybamm/config.py

```diff
@@ -0,0 +1,141 @@
+import uuid
+import os
+import platformdirs
+from pathlib import Path
+import pybamm
+from inputimeout import inputimeout, TimeoutOccurred
+
+
+def is_running_tests():  # pragma: no cover
+    """
+    Detect if the code is being run as part of a test suite or building docs with Sphinx.
+
+    Returns:
+        bool: True if running tests or building docs, False otherwise.
+    """
+    import sys
+
+    # Check if pytest or unittest is running
+    if any(
+        test_module in sys.modules for test_module in ["pytest", "unittest", "nose"]
+    ):
+        return True
+
+    # Check for GitHub Actions environment variable
+    if "GITHUB_ACTIONS" in os.environ:
+        return True
+
+    # Check for other common CI environment variables
+    ci_env_vars = ["CI", "TRAVIS", "CIRCLECI", "JENKINS_URL", "GITLAB_CI"]
+    if any(var in os.environ for var in ci_env_vars):
+        return True
+
+    # Check for common test runner names in command-line arguments
+    test_runners = ["pytest", "unittest", "nose", "trial", "nox", "tox"]
+    if any(runner in sys.argv[0].lower() for runner in test_runners):
+        return True
+
+    # Check if building docs with Sphinx
+    if "sphinx" in sys.modules:
+        return True
+
+    return False
+
+
+def ask_user_opt_in(timeout=10):
+    """
+    Ask the user if they want to opt in to telemetry.
+
+    Parameters
+    ----------
+    timeout: float, optional
+        The timeout for the user to respond to the prompt. Default is 20 seconds.
```

Comment:
Default above appears to be 10 seconds
```suggestion
        The timeout for the user to respond to the prompt. Default is 10 seconds.
```